constructor_io/constructor_io.py

```diff
@@ -45,24 +66,44 @@ def _make_url(self, endpoint, params=None):
         return "{0}://{1}/{2}?{3}".format(self._protocol, self._host, endpoint,
                                           self._serialize_params(params))
 
-    def query(self, query_str):
+    def __make_server_request(self, request_method, *args, **kwargs):
+        retries_left = self._server_error_retries
+
+        while True:
+            try:
+                # Wrap server error codes as exceptions
+                response = request_method(*args, **kwargs)
+                if round(response.status_code / 100) == 5:
```

Comment:
I believe that integer division would be more readable here. That is instead of:
`round(response.status_code / 100) == 5`

this:
`(response.status_code // 100) == 5`

Reply:
Also, note that `round(451 / 100)` is `5`, not `4`.

Reply:
Agree about integer division here. By the way: ```
$ python
Python 2.7.6 (default, Jun 22 2015, 17:58:13) 
[GCC 4.8.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> round(451 / 100)
4.0
>>> round(459 / 100)
4.0
>>> round(499 / 100)
4.0
>>> 
```

Reply:
My bad, I checked with python 3. Anyway, from what I see, the code is expected to work with both python 2 and 3, so it's better not to use `round` just in case.